src/sentry/workflow_engine/handlers/action/notification.py

```diff
@@ -1,17 +1,54 @@
+import logging
+from typing import Protocol
+
+from sentry.grouping.grouptype import ErrorGroupType
+from sentry.issues.grouptype import MetricIssuePOC
+from sentry.utils.registry import NoRegistrationExistsError, Registry
 from sentry.workflow_engine.models import Action, Detector
 from sentry.workflow_engine.registry import action_handler_registry
 from sentry.workflow_engine.types import ActionHandler, WorkflowJob
 
+logger = logging.getLogger(__name__)
+
+
+class NotificationHandler(Protocol):
+    """
+    A protocol that defines the interface for the method that will invoke the existing notification registry.
+    """
+
+    def __call__(self, job: WorkflowJob, action: Action, detector: Detector) -> None: ...
+
+
+# Register what function to use based on the group type
+notification_registry = Registry[NotificationHandler]()
 
-# TODO - Enable once the PR to allow for multiple of the same funcs is merged
-# @action_handler_registry.register(Action.Type.PAGERDUTY)
-@action_handler_registry.register(Action.Type.SLACK)
+
+@action_handler_registry.register(Action.Type.DISCORD)
 class NotificationActionHandler(ActionHandler):
     @staticmethod
     def execute(
         job: WorkflowJob,
         action: Action,
         detector: Detector,
     ) -> None:
-        # TODO: Implment this in milestone 2
-        pass
+        try:
+            handler = notification_registry.get(detector.type)
```

Comment:
You should be able to mock this registry `get` call in order to test this logic. You can also assert that the mock get call is invoked with the correct detector type if you'd like.